tests/kernel/threads/usage/src/task.c

```diff
@@ -0,0 +1,384 @@
+	k_sem_take(&sem_thread2, K_FOREVER);
+	thread2_data = k_thread_priority_get(k_current_get());
+	k_sem_give(&sem_thread1);
+}
+
+/**
+ *
+ * @brief Test the k_thread_priority_set() API
+ *
+ */
+
+void thread_set_prio_test(void)
+{
+	int  rv;
+
+	/* Lower the priority of the current thread (thread1) */
+	k_thread_priority_set(thread1_id, THREAD1_PRIO + 2);
+	rv = k_thread_priority_get(k_current_get());
+	zassert_equal(rv, THREAD1_PRIO + 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD1_PRIO + 2, rv);
+
+	/* Raise the priority of the current thread (thread1) */
+	k_thread_priority_set(thread1_id, THREAD1_PRIO - 2);
+	rv = k_thread_priority_get(k_current_get());
+	zassert_equal(rv, THREAD1_PRIO - 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD1_PRIO - 2, rv);
+
+	/* Restore the priority of the current thread (thread1) */
+	k_thread_priority_set(thread1_id, THREAD1_PRIO);
+	rv = k_thread_priority_get(k_current_get());
+	zassert_equal(rv, THREAD1_PRIO,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD1_PRIO, rv);
+
+	/* Lower the priority of thread2 */
+	k_thread_priority_set(thread2_id, THREAD2_PRIO + 2);
+	k_sem_give(&sem_thread2);
+	k_sem_take(&sem_thread1, K_FOREVER);
+	zassert_equal(thread2_data, THREAD2_PRIO + 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD2_PRIO + 2, thread2_data);
+
+	/* Raise the priority of thread2 */
+	k_thread_priority_set(thread2_id, THREAD2_PRIO - 2);
+	k_sem_give(&sem_thread2);
+	k_sem_take(&sem_thread1, K_FOREVER);
+	zassert_equal(thread2_data, THREAD2_PRIO - 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD2_PRIO - 2, thread2_data);
+
+	/* Restore the priority of thread2 */
+	k_thread_priority_set(thread2_id, THREAD2_PRIO);
+	k_sem_give(&sem_thread2);
+	k_sem_take(&sem_thread1, K_FOREVER);
+	zassert_equal(thread2_data, THREAD2_PRIO,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD2_PRIO, thread2_data);
+}
+
+/**
+ *
+ * @brief thread2 portion to test k_sleep()
+ *
+ */
+
+void thread2_sleep_test(void)
+{
+	int  first_tick;
+
+	k_sem_take(&sem_thread2, K_FOREVER);
+
+	first_tick = _tick_get_32();
+	while (!is_main_thread_ready) {
+		/* busy work */
+	}
+	thread2_data = _tick_get_32() - first_tick;
+
+	k_sem_give(&sem_thread1);
+}
+
+/**
+ *
+ * @brief Test k_sleep()
+ *
+ */
+
+void thread_sleep_test(void)
+{
+	int  tick;
+
+	k_sem_give(&sem_thread2);
+
+	/* align on tick boundary and get current tick */
+	tick = _tick_get_32();
+	while (tick == _tick_get_32()) {
+	}
+
+	/* compensate for the extra tick we just waited */
+	++tick;
+
+	k_sleep(SLEEP_TIME);
+
+	tick = _tick_get_32() - tick;
+
+	is_main_thread_ready = 1;
+	k_sem_take(&sem_thread1, K_FOREVER);
+
+	/*
+	 * By design this should be exact, but at least one cycle of
+	 * slop is required experimentally on Qemu.
+	 */
+	zassert_true(tick < SLEEP_TIME || tick > SLEEP_TIME + 1,
```

Comment:
you are comparing ticks with msec. you will have to do
__ticks_to_ms(tick)

also don't use  _tick_get_32() as it is kernel internal function, you can look into k_uptime_get_32() as it returns uptime in msec.